### PR TITLE
For supporting redis authentication. Add support api for opening connection with redis client as parameter

### PR DESCRIPTION
**Pull request by anupamSimility**

### CODE REVIEW

**Comment by anupamSimility** [comment]
Using password protected redis. No api with password is exposed. Adding api which will pass redis client as parameter. It can be build with Options (https://github.com/adjust/redis/blob/master/redis.go#L154)

**Comment by anupamSimility** [comment]
If not, can you please expose api with password

func OpenConnection(tag, network, address, password string, db int) *redisConnection{
redisClient := redis.NewClient(&redis.Options{
		Network: network,
		Addr:    address,
		DB:      int64(db),
                Password: password,
	})

...

return connection;
}

File: connection.go [hunk 1]
```diff
@@ -29,14 +29,8 @@ type redisConnection struct {
...
 
-// OpenConnection opens and returns a new connection
-func OpenConnection(tag, network, address string, db int) *redisConnection {
-	redisClient := redis.NewClient(&redis.Options{
-		Network: network,
-		Addr:    address,
-		DB:      int64(db),
-	})
-
+//OpenConnectionWithRedisClient opens and return a new connection
```
Comment (wellle) [inline connection.go]: Could you add a space after `//` and use `returns` instead of `return`?

Comment (anupamSimility) [inline connection.go]: done.

**Comment by wellle** [comment]
Thank you :+1:

**Pull request merged**